pkg/documents/documents.go

```diff
@@ -184,21 +202,20 @@ func retrieveRawDocument(logger log.Logger, repo DocumentRepository, bucketFacto
 		ctx, cancelFn := context.WithTimeout(context.TODO(), 10*time.Second)
 		defer cancelFn()
 
-		documentKey := makeDocumentKey(customerID, documentId)
-		signedURL, err := bucket.SignedURL(ctx, documentKey, &blob.SignedURLOptions{
-			Expiry: 15 * time.Minute,
-		})
+		documentKey := makeDocumentKey(customerID, documentID)
+		rdr, err := bucket.NewReader(ctx, documentKey, nil)
 		if err != nil {
-			moovhttp.Problem(w, err)
+			moovhttp.Problem(w, fmt.Errorf("error reading %s document: %v", documentKey, err))
 			return
 		}
-		if signedURL == "" {
-			moovhttp.Problem(w, fmt.Errorf("document=%s not found", documentId))
+		defer rdr.Close()
+
+		w.Header().Set("Content-Type", rdr.ContentType())
+		w.WriteHeader(http.StatusOK)
+		if n, err := io.Copy(w, rdr); n == 0 || err != nil {
+			moovhttp.Problem(w, fmt.Errorf("problem writing %s (n=%d) document: %v", documentKey, n, err))
```

Comment:
`n` is somewhat unclear in the error message here since its context comes from the underlying code. Something like `# of bytes copied = %d` would work well.